src/tests/posix/pthread/pthread_barrier_test.c

```diff
@@ -0,0 +1,43 @@
+/**
+ * @file
+ * @brief   Test pthread_barrier_functions
+ * @date    Mar 12, 2020
+ * @author  Sobhan Mondal <inro20fdm@gmail.com>
+ */
+
+#include <pthread.h>
+#include <embox/test.h>
+#include <unistd.h>
+#include <stdlib.h>
+
+EMBOX_TEST_SUITE("posix/pthread_barrier_api");
+
+TEST_CASE ("Initialize barrier with pthread_barrier_init()") {
+	pthread_barrier_t barrier = NULL; 
+	pthread_barrierattr_t attr = (pthread_barrierattr_t) malloc(sizeof(*attr));
+	attr->pshared = PTHREAD_PROCESS_PRIVATE;
```

Comment:
As I can see from the standard (https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html), `pthread_barrierattr_t` is not explicitly required to be a pointer, so test should not rely on fact that it is a pointer, i.e. `sizeof(*attr)` may fail to compile if implementation will change (but will still be POSIX-compliant).

Also it means that you can't access `attr->pshared` directly.

Reply:
Okay! [According to this](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html), `pthread_barrier_t` also should not be a pointer type.  Right?

Reply:
It's unspecified, so it's implementation-dependent.